client/src/components/Ticketing/ticketingmanager/AdminPurchase/ReaderPurchase.tsx

```diff
@@ -0,0 +1,234 @@
+import React, {useEffect, useState} from 'react';
+import {
+  Button,
+} from '@mui/material';
+import PopUp from '../../PopUp';
+import {useNavigate, useParams, useLocation} from 'react-router-dom';
+import useWebSocket from 'react-use-websocket';
+
+import {useAppDispatch, useAppSelector} from '../../app/hooks';
+
+import {useFetchToken} from '../Event/components/ShowingUtils';
+
+import {
+  removeAllTicketsFromCart,
+  selectDiscount,
+} from '../ticketingSlice';
+
+import {loadStripe} from '@stripe/stripe-js';
+
+const pk = `${process.env.REACT_APP_PUBLIC_STRIPE_KEY}`;
+const stripePromise = loadStripe(pk);
+
+const ReaderPurchase = () => {
+  const {token} = useFetchToken();
+  const [status, setStatus] = useState('Awaiting Response...');
+
+  const location = useLocation();
+  const navigate = useNavigate();
+  const cartItems = location.state?.cartItems || [];
+  const clientSecret = location.state?.clientSecret || '';
+  const readerID = location.state?.readerID || '';
+  const discount = useAppSelector(selectDiscount);
+  const [openDialog, setDialog] = useState(false);
+  const [errMsg, setErrMsg] = useState('');
+
+  const event = 'reader';
+  const paymentIntentID = useParams().id;
+
+  const socketURL = process.env.REACT_APP_WEBSOCKET_URL;
+
+  const {getWebSocket} = useWebSocket(socketURL, {
+    shouldReconnect: () => true,
+    onMessage: (event) => {
+      const ws = getWebSocket();
+      const data = JSON.parse(event.data);
+      if (data.messageType === 'reader' &&
+          data.paymentIntent === paymentIntentID ||
+          data.paymentIntent === readerID) { // in the case of terminal event, paymentIntent is a readerID
+        setStatus(data.eventType);
+        if (data.eventType === 'payment_intent.succeeded') {
+          ws.close();
+          navigate(`/success`);
+        } else if (data.eventType === 'payment_intent.requires_action') { // not sure if this needs to be handelled but I think this is how its done
+          stripePromise.then((stripe) => {
+            if (!stripe) return;
+            stripe.confirmCardPayment(clientSecret).then((result) => {
+              console.log(result); // should handle failure/success here
```

Comment:
What is _supposed_ to happen here?

Reply:
This was a misunderstanding I had earlier that I had meant to remove, our payment intent object actually has automatic confirmation enabled, and confirmCardPayment only applies to web purchases because it redirects you to a confirmation page, this case has been removed.